src/python/pants/rules/core/list_roots_test.py

```diff
@@ -77,6 +77,36 @@ def provider_rule(path_globs: PathGlobs) -> Snapshot:
             set(output),
         )
 
+    def test_all_roots_with_root_at_buildroot(self):
+        options = {
+            "pants_ignore": [],
+            "source_root_patterns": [],
+            "source_roots": {"": ["python"],},
+        }
+        options.update(self.options[""])  # We need inherited values for pants_workdir etc.
+
+        self.context(
+            for_subsystems=[SourceRootConfig], options={SourceRootConfig.options_scope: options}
+        )
+
+        source_root_config = SourceRootConfig.global_instance()
+        source_roots = source_root_config.get_source_roots()
+
+        # This function mocks out reading real directories off the file system
+        def provider_rule(path_globs: PathGlobs) -> Snapshot:
+            dirs = ("foo",)  # A python package at the buildroot.
+            return Snapshot(Digest("abcdef", 10), (), dirs)
+
+        output = run_rule(
+            list_roots.all_roots,
+            rule_args=[source_root_config],
+            mock_gets=[MockGet(product_type=Snapshot, subject_type=PathGlobs, mock=provider_rule),],
```

Comment:
The trailing commas in this file will cause Flake8 to fail now that we merged turning on the lint. They need to be manually removed.

Reply:
Then we should turn off the lint, probably. It is double-plus-ungood if Black cannot format our code correctly without manual fixes. The whole point of Black is "whatever it does is what is correct, by fiat".

Reply:
+1 to turning off the lint. You can't adopt an autoformatter and then complain about its formatting.

Reply:
Okay. I'll turn it off :/ I agree with you both. Hopefully Black changes this behavior.